src/awsHttpHandler/httpHandler.ts

```diff
@@ -0,0 +1,116 @@
+import { hckFetch } from '../index';
+import { requestTimeout } from './requestTimeout';
+import { isEmpty } from './isEmpty';
+
+type LogHandler = (...content: unknown[]) => string;
+
+export type Logger = {
+  log?: LogHandler;
+  info?: LogHandler;
+};
+
+export type HttpHandlerConstructor = {
+  requestTimeout?: number;
+  logger?: Logger;
+};
+
+type RequestOptions = {
+  headers: Headers;
+  body?: any;
+  method: string;
+  signal?: AbortSignal;
+};
+
+export class HttpHandler {
+  requestTimeout?: HttpHandlerConstructor['requestTimeout'];
+  logger?: Logger;
```

Comment:
I suggest setting the logger to a no-op version by default (aka log functions that exist but do nothing). That way, you don't need to deal with the logger being `undefined` later in your logic.

```suggestion
  logger: Logger;
```